docs/design-documents/tools/cmake.md

```diff
@@ -1,101 +1,95 @@
 
-There are 3 toolchains, we provide the template that tools could fill in. The information is already in the build tools, we just need to extract that info and make it CMake compatible.
+All the toolchain settings are defined in the scripts found in `cmake/toolchains/`.
 
-This toolchain CMake is included by the Mbed OS CMake.
+### 3. Profile CMake
 
-### 3. Module CMake
+The build profiles such as release or debug are defined in the scripts found in `cmake/profiles/`.
 
-This file statically defines the structure of the module within Mbed OS. It's conditionally config based. We use regular CMake expressions plus extend it with own macros to conditionally include/exclude directories. The thumb of the rule, do not expose headers that are internal. We would like to avoid having everything in the include paths as we do now.
+### 4. MCU core CMake
 
+The MCU core definitions are defined in the scripts found in `cmake/cores/`.
 
-`add_subdirectory` always adds the directory to the main CMake.
+### 5. Utilities CMake
 
-```
-add_subdirectory(drivers)
-```
+Custom functions/macros used within Mbed OS.
 
-Conditionally directories addition based on the config:
-
-```
-add_subdirectory_if_config(CONFIG_REQUIRES_RTOS rtos)
-```
-
-Conditionally directories addition based on the target/toolchain/feature.
+### 6. Application CMake
 
+The CMake script that must be included by all applications using:
 
 ```
-add_subdirectory_if_target(STM targets\TARGET_STM)
-add_subdirectory_if_toolchain(GCC_ARM cmsis\TARGET_GCC_ARM)
-add_subdirectory_if_feature(BLE ble\FEATURE_BLE)
+include(${MBED_ROOT}/cmake/app.cmake)
 ```
 
+### 7. Component CMake
+
+This file statically defines the structure of an Mbed OS component. It also contains conditional statements that are based on the configuration. Regular CMake expressions and Mbed OS functions/macros are used in it to conditionally include/exclude directories.
+The rule of thumb is to not expose header files that are internal. We would like to avoid having everything in the include paths as we do now.
 
-## Application CMake
+## Building an application
 
-We should provide application CMake functionality with  our own configuration. There are couple of approaches we could take. Statically defined CMake but then this disconnectes config and CMake - as CMake contains configuration for a project (like includes, sources, etc). Our build tool would need to parse CMake to get all paths used in the project or Mbed OS to find out where to look for configuration file. Therefore the build system has a knowledge as it is currently. We use `requires` to include/exclude modules.
+mbed-tools is the next generation mbed-cli. It replaces the current mbed-cli and Mbed OS tools (build system part of Mbed OS).
 
-By default, baremetal would be selected - requires set to hal, platform, drivers and cmsis. If an app needs anything else, would use `requires` in the config to include - BLE/networking/etc.
+TBD?:
+- configuration
```

Comment:
```suggestion
### Configuration

The main purpose of `mbed-tools` is to parse the Mbed configuration system's JSON files (`mbed_lib.json`, `mbed_app.json` and `targets.json`). The tool outputs a single CMake configuration script, which is included by `app.cmake` and `mbed-os/CMakeLists.txt`.

To generate the CMake config script (named `mbed_config.cmake`) the user can run the `configure` command:

`mbedtools configure -t <toolchain> -m <target>`

This will output `mbed_config.cmake` in a directory named `.mbedbuild` at the root of the program tree.

`mbed_config.cmake` contains several variable definitions used to select the toolchain, core and profile CMake scripts to be used in the build system generation:
* `MBED_TOOLCHAIN`
* `MBED_TARGET`
* `MBED_CPU_CORE`
* `MBED_PROFILE`

The tools also generate an `MBED_TARGET_LABELS` variable, containing the labels, components and feature definitions from `targets.json`, used to select the required Mbed OS components to be built.

The macro definitions parsed from the Mbed OS configuration system are also included in `mbed_config.cmake`. The decision was made to remove `mbed_config.h`.
```